usersession/agent/rest_api.go

```diff
@@ -460,11 +462,37 @@ func postRefreshFinishedNotification(c *Command, r *http.Request) Response {
 		notification.WithDesktopEntry("io.snapcraft.SessionAgent"),
 		notification.WithUrgency(notification.LowUrgency),
 	}
+	si, _ := snap.ReadCurrentInfo(finishRefresh.InstanceName)
```

Comment:
If you discard the error, then si could be nil, which means the code below will panic in such scenario
```suggestion
	si, err := snap.ReadCurrentInfo(finishRefresh.InstanceName)
	if err == nil {
		// do the happy path processing, find the app etc.
	} else {
		logger.Noticef("cannot load snap-info: %v", err)
	}	 
```

note you may need to import logger

Reply:
Good point!

Reply:
Done. Also, now it really takes the first icon found instead of the last one (added a check when getting the icon name, and a break if one is found).